cumulus/parachains/runtimes/assets/asset-hub-westend/src/lib.rs

```diff
@@ -1450,31 +1450,31 @@ impl_runtime_apis! {
 			// We accept the native token to pay fees.
 			let mut acceptable_assets = vec![AssetId(native_token.clone())];
 			// We also accept all assets in a pool with the native token.
-			acceptable_assets.extend(
-				pallet_asset_conversion::Pools::<Runtime>::iter_keys().filter_map(
-					|(asset_1, asset_2)| {
-						if asset_1 == native_token {
-							Some(asset_2.clone().into())
-						} else if asset_2 == native_token {
-							Some(asset_1.clone().into())
-						} else {
-							None
-						}
-					},
-				),
-			);
+			acceptable_assets.extend(assets_common::get_assets_in_pool_with::<Runtime>(&native_token).map(|location| AssetId(location)));
```

Comment:
not sure, does this also compile?
```suggestion
			acceptable_assets.extend(assets_common::get_assets_in_pool_with::<Runtime>(&native_token).map(AssetId));
```